src/domain/auth/authService.ts

```diff
@@ -79,22 +79,20 @@ class AuthService {
     return this.userManager.signinRedirect({ data: { path } });
   }
 
-  public endLogin(): Promise<User> {
-    return this.userManager.signinRedirectCallback().then(async user => {
-      this.fetchApiTokens(user);
-
-      return user;
-    });
+  public async endLogin(): Promise<User> {
+    const user = await this.userManager.signinRedirectCallback();
+    await this.fetchApiTokens(user);
+    return user;
   }
 
   public renewToken(): Promise<User> {
     return this.userManager.signinSilent();
   }
 
-  public logout(): Promise<void> {
-    return this.userManager.signoutRedirect().then(() => {
-      localStorage.removeItem(API_TOKENS);
-    });
+  public async logout(): Promise<void> {
+    await this.userManager.signoutRedirect();
```

Comment:
I think there is still a possibility for a race condition here. `signoutRedirect` will redirect to another url. So if the redirection is fast enough the JS in our site may not be executed. Could we change the sequence as follows:
```suggestion
    localStorage.removeItem(API_TOKENS);
    this.userManager.clearStaleState();
    await this.userManager.signoutRedirect();
```

Reply:
Good point, fixed!